aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -237,6 +235,62 @@ Tensor _embedding_bag_backward(const Tensor &grad, const Tensor &indices,
   }
 }
 
+static Tensor _embedding_bag_dense_backward_cpu_max(
+    const Tensor& grad,
+    const Tensor& bag_size,
+    const Tensor& max_indices,
+    int64_t num_weights) {
+  AT_ASSERT(max_indices.defined());
+  auto index_grad_weight =
+      at::zeros({num_weights, grad.size(1)}, grad.options());
+  auto nonempty_max_indices = max_indices.index_select(0, bag_size.nonzero().view(-1));
+  auto nonempty_grad = grad.index_select(0, bag_size.nonzero().view(-1));
+
+  for (int64_t dim = 0; dim < grad.size(1); dim++) {
+    index_grad_weight.select(1, dim).index_add_(
+      0, nonempty_max_indices.select(1, dim), nonempty_grad.select(1, dim));
+  }
+  return index_grad_weight;
+}
+
+static std::vector<int64_t> compute_counts(
```

Comment:
I'm assuming this is just a copy-paste of a block of code previously part of _embedding_bag_dense_backward_cpu.

Reply:
yes